src/utils.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Server.hpp"
+
+void trim( std::string& str ) {
```

Comment:
You can use the `find_first_not_of()` and `find_last_not_of()` to find where does the actual string starts so the code looks more simple.
```C++
void trim( std::string& str ) {
    const std::string set = " "; // add all the unwanted chars
    size_t start = str.find_first_not_of( set );
    size_t end = str.find_last_not_of( set );
    str = str.substr( start, end - start + 1 );
}
```

Reply:
did not think of that, will be implementing this in the next pull request